include/pistache/async.h

```diff
@@ -203,8 +205,12 @@ namespace Async {
                 }
 
                 void *mem = memory();
+                if (is_construct) {
+                    reinterpret_cast<T *>(mem)->~T();
+                };
                 new (mem) T(std::forward<Args>(args)...);
```

Comment:
Actually, this is not exception safe. It would be probably better like that:
```cpp
auto new_mem = std::make_unique<T>(std::forward<Args>(args)...); // new may throw here
void* mem = memory();
if (is_construct && mem != nullptr)
    reinterpret_cast<T*>(mem)->~T();
mem = new_mem.release();
state = State::Fulfilled;
is_construct = true;
```